client/client.go

```diff
@@ -302,12 +302,28 @@ func addChange(cl *changelist.FileChangelist, c changelist.Change, roles ...stri
 	return nil
 }
 
-// AddDelegation creates a new changelist entry to add a delegation to the repository
-// when the changelist gets applied at publish time.  This does not do any validation
-// other than checking the name of the delegation to add - all that will happen
-// at publish time.
-func (r *NotaryRepository) AddDelegation(name string, threshold int,
-	delegationKeys []data.PublicKey, paths []string) error {
+// AddDelegation creates changelist entries to add provided delegation public keys and paths.
+// This method composes AddDelegationRoleAndKeys and AddDelegationPaths (each creates one changelist if called).
+func (r *NotaryRepository) AddDelegation(name string, delegationKeys []data.PublicKey, paths []string) error {
```

Comment:
I'm fine adding back in the `threshold` argument both here and `AddDelegationRoleAndKeys`.  I originally didn't include `threshold` in these APIs because we never set them to any other value but 1 in our code, nor do we currently have all of the thresholding logic completely implemented.

We can future-proof this method by keeping the threshold argument, or we can leave as is since the change files that these APIs produce are still processed in the same fashion.  Either way is fine with me, @endophage @cyli WDYT?